src/ml/neural_net/tf_compute_context.cpp

```diff
@@ -43,6 +41,23 @@ class tf_model_backend : public model_backend {
   pybind11::object model_;
 };
 
+class tf_image_augmenter : public processed_image_augmenter {
+ public:
+  tf_image_augmenter(const options& opts);
+  ~tf_image_augmenter();
```

Comment:
Since you're using inheritance, should the destructor be virtual?

Reply:
The idiom to follow here is
```
~tf_image_augmenter() override = default;
```

This asserts that (a) the parent class has a virtual destructor, and (b) this class doesn't introduce any additional logic for when an instance is destroyed.

Then you can omit the empty definition below